common/src/main/java/org/opensearch/ml/common/transport/agent/MLAgentUpdateInput.java

```diff
@@ -0,0 +1,288 @@
+                    break;
+                case TOOLS_FIELD:
+                    tools = new ArrayList<>();
+                    ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.currentToken(), parser);
+                    while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                        tools.add(MLToolSpec.parse(parser));
+                    }
+                    break;
+                case PARAMETERS_FIELD:
+                    parameters = parser.mapStrings();
+                    break;
+                case MEMORY_FIELD:
+                    memory = MLMemorySpec.parse(parser);
+                    break;
+                case APP_TYPE_FIELD:
+                    appType = parser.text();
+                    break;
+                case LAST_UPDATED_TIME_FIELD:
+                    lastUpdateTime = Instant.ofEpochMilli(parser.longValue());
+                    break;
+                case TENANT_ID_FIELD:
+                    tenantId = parser.textOrNull();
+                    break;
+                default:
+                    parser.skipChildren();
+                    break;
+            }
+        }
+
+        return new MLAgentUpdateInput(agentId, name, description, llm, tools, parameters, memory, appType, lastUpdateTime, tenantId);
+    }
+
+    public MLAgent toMLAgent(MLAgent originalAgent) {
+        return MLAgent
+            .builder()
+            .type(originalAgent.getType())
+            .createdTime(originalAgent.getCreatedTime())
+            .isHidden(originalAgent.getIsHidden())
+            .name(name == null ? originalAgent.getName() : name)
+            .description(description == null ? originalAgent.getDescription() : description)
+            .llm(llm == null ? originalAgent.getLlm() : llm)
+            .tools(tools == null ? originalAgent.getTools() : tools)
+            .parameters(parameters == null ? originalAgent.getParameters() : parameters)
+            .memory(memory == null ? originalAgent.getMemory() : memory)
+            .lastUpdateTime(lastUpdateTime)
+            .appType(appType)
+            .tenantId(tenantId)
+            .build();
+    }
+
+    private void validate() {
+        if (name != null && (name.isBlank() || name.length() > MLAgent.AGENT_NAME_MAX_LENGTH)) {
+            throw new IllegalArgumentException(
+                String.format("Agent name cannot be empty or exceed max length of %d characters", MLAgent.AGENT_NAME_MAX_LENGTH)
+            );
+        }
+        if (memory != null && !memory.getType().equals("conversation_index")) {
+            throw new IllegalArgumentException(String.format("Invalid memory type: %s", memory.getType()));
+        }
+        if (tools != null) {
+            Set<String> toolNames = new HashSet<>();
+            for (MLToolSpec toolSpec : tools) {
+                String toolName = Optional.ofNullable(toolSpec.getName()).orElse(toolSpec.getType());
+                if (toolNames.contains(toolName)) {
```

Comment:
Do we need to throw the exception for duplicate? Can't we just ignore? What's the requirement here?